hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1329,8 +1329,16 @@ public OmMultipartUploadListParts listParts(String volumeName,
             multipartKeyInfo.getPartKeyInfoMap();
         Iterator<Map.Entry<Integer, PartKeyInfo>> partKeyInfoMapIterator =
             partKeyInfoMap.entrySet().iterator();
-        HddsProtos.ReplicationType replicationType =
-            partKeyInfoMap.firstEntry().getValue().getPartKeyInfo().getType();
+
+        OmKeyInfo omKeyInfo =
+            metadataManager.getOpenKeyTable().get(multipartKey);
+
+        if (omKeyInfo == null) {
+          throw new IllegalStateException(
+              "Open key is missing for multipart upload " + multipartKey);
+        }
+
+        HddsProtos.ReplicationType replicationType = omKeyInfo.getType();
```

Comment:
We are setting replicationType twice- line 1358 and 1341. Can we add this openKeyTable get op inside an if condition so that it is executed only if no parts are present in the key?

Reply:
Yes agreed, if we do this way, we can save one DB read.

Reply:
Thanks the idea @lokeshj1703 and @bharatviswa504 

Updated the patch according to this suggestion.